pkg/selector/selector.go

```diff
@@ -171,15 +176,35 @@ func (itf Selector) rawFilter(filters Filters) ([]*ec2.InstanceTypeInfo, error)
 	}
 
 	instanceTypesInput := &ec2.DescribeInstanceTypesInput{}
-	instanceTypeCandidates := map[string]*ec2.InstanceTypeInfo{}
+	instanceTypeCandidates := map[string]*instancetypes.Details{}
 	// innerErr will hold any error while processing DescribeInstanceTypes pages
 	var innerErr error
 
 	err = itf.EC2.DescribeInstanceTypesPages(instanceTypesInput, func(page *ec2.DescribeInstanceTypesOutput, lastPage bool) bool {
 		for _, instanceTypeInfo := range page.InstanceTypes {
 			instanceTypeName := *instanceTypeInfo.InstanceType
-			instanceTypeCandidates[instanceTypeName] = instanceTypeInfo
+			instanceTypeCandidates[instanceTypeName] = &instancetypes.Details{InstanceTypeInfo: *instanceTypeInfo}
 			isFpga := instanceTypeInfo.FpgaInfo != nil
+			instanceTypeHourlyPrice := float64(0.0)
+			if filters.PricePerHour != nil {
+				if filters.UsageClass != nil && *filters.UsageClass == "spot" {
+					azs := []string{}
+					if filters.AvailabilityZones != nil {
+						azs = *filters.AvailabilityZones
+					}
+					instanceTypeHourlyPrice, err = itf.EC2Pricing.GetSpotInstanceTypeNDayAvgCost(instanceTypeName, azs, 30)
+					if err != nil {
+						fmt.Printf("Could not retrieve 30 day avg spot price for instance type %s\n", instanceTypeName)
```

Comment:
nit: error msg should be *Could not retrieve N day avg spot price*

Reply:
👍  hmm I guess I hardcoded the lookup to 30 days... that should probably be configurable.. but I think I'd opt to leave it at 30 for now and see if we need it more configurable since there's already a ton of configuration...

Reply:
I saw the func name and assumed it was going to be configurable via cli arg, but that's not necessarily true. it's good as-is until that functionality is added, if ever